### ISSUE TITLE
M3U plugin failing to parse m3u

### ISSUE THREAD

**Issue by subby2006** [cabernetwork/provider_video_m3u#8]
Currently trying to configure a custom m3u for the locals but its failing to parse due to a missing #EXTM3U even though it's clearly there in the file
```
2024-08-21 04:29:47,055-NOTICE:scheduler Running inline task Channels:Refresh M3U Instance Channels
2024-08-21 04:29:47,149-WARNING:__init__ INVALID m3u format: #EXTM3U missing /app/data/tmp/m3u/1724214587.0864246.m3u
2024-08-21 04:29:47,150-WARNING:scheduler CabernetException: M3U Channel Request Failed, no M3U data in the file for instance default
```
my configuration in config.ini is this 
```
hdhr_id = 105A14D1
enabled = True
label = m3u_locals
channel-m3u_file = my m3u url
stream-g_http_timeout = 5
stream-g_http_retries = 2
player-send_atsc_keepalive = True
```

and here is whats at the top of my m3u playlist
![image](https://github.com/user-attachments/assets/9eda8c4f-e5f9-468d-8ce5-18fc37c86181)

could you help out here? i can give the link to my playlist if you want

**Comment by rocky4546** [comment]
(edited) The error must be correct, so the plan is to find out what is really being sent. Try this to print out in the log what is the file
Edit channels.py in the m3u plugin lib folder.  Add the following after line 73 (73 included here) to print out the data.
```
            m3u_file = self.extract_file(dn_filename, file_type)
            self.logger.warning('######################')
            self.logger.warning(m3u_file)
            if m3u_file is not None:
               f = open(m3u_file, "r")
               self.logger.warning(f.read()) 
               f.close()
            self.logger.warning('######################')
```

**Comment by subby2006** [comment]
Alright I seemed to have figured out the problem, it was a duplicate entry that was causing the problem. Cabernet was able to get my playlist however I seemed to have run into a second issue, the streams aren’t being proxied by any of the software, be it ffmpeg, streamlink or internal and I suspect it’s due to how the provider is streaming it, which is through an mpeg-ts stream instead of an HLS playlist. Any chance support for mpeg-ts streams would be given?

**Comment by rocky4546** [comment]
No, mpeg-ts works fine.  You have not provided much info, but if I was to take a wild guess, I would guess the issue is the TS files are protected by a Referrer.  The M3U file can state the Referrer required, but again that is a guess.

**Comment by subby2006** [comment]
They aren't protected by a referrer, I'm able to play them fine in VLC. Here is my playlist for reference
https://gist.githubusercontent.com/subby2006/e6c8c83f8212e38b11a3365dd3391b1a/raw/b7700b11eb09873c95b8b3a867ddb472b71abef0/cabernet.m3u

**Comment by rocky4546** [comment]
It works for ffmpegproxy and m3u8redirect, but will not work for streamlink or internal.  This is because those two expect a non-continuous url packet.  In this case, it is a continuous stream.  To get ffmpegproxy or m3u8proxy to work, you need to create some files.

For each channel, create a file (ex: USA-CW.m3u) containing something like:

#EXTM3U
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=1042180,SUBTITLES="subs"
http://smartnow.tk:8080/ana7922/qwe123321/4800

For now, place these files in the data folder of cabernet.

Then in the master file (cabernet.m3u), you can list the files like so,

#EXTM3U
#EXTINF:-1,USA- THE CW HD - CHICAGO
../../USA-CW.m3u8

This should then pick up the url and throw it to the process for streaming.  you can create a subfolder in data and point to that if you want.

**Comment by subby2006** [comment]
Everything seems to be working now however there is a small issue. At around 1 minute to 1:30 of staying on the stream, the stream locks up for around 2-3 minutes and I see these in the console
```
2024-08-25 03:30:27,004-INFO:ffmpeg_proxy No Video Data, refreshing stream 243
2024-08-25 03:30:30,465-INFO:ffmpeg_proxy Serving 255 1 (149272B) ttw:0.00s
2024-08-25 03:30:31,569-INFO:ffmpeg_proxy 1. Connection dropped by end device 259
2024-08-25 03:30:33,194-NOTICE:web_tuner Provider Connection Closed, ch_id=1 140503158496056
2024-08-25 03:32:31,632-INFO:ffmpeg_proxy No Video Data, refreshing stream 255
2024-08-25 03:32:43,973-INFO:ffmpeg_proxy Serving 263 1 (3539100B) ttw:0.00s
2024-08-25 03:32:45,177-INFO:ffmpeg_proxy Serving 263 1 (227668B) ttw:0.00
```
After that the stream resumes and is rock solid with no disconnects or drops for the entire duration. This issue also happens when another client joins the stream, at which both the connections are dropped for around 2 minutes before resuming. Cabernet doesn't disconnect or send an EOF to the player, its just frozen until the next video packet. Not sure if its the video host acting wonky or not.
